roles/servicetelemetry/tasks/component_servicemonitor.yml

```diff
@@ -27,14 +27,26 @@
               - action: labeldrop
                 regex: publisher
                 sourcelabels: []
-            port: prom-http
+            port: prom-https
+            scheme: https
+            tlsConfig:
+              caFile: /etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt
+              serverName: "{{ this_smartgateway }}.{{ meta.namespace }}.svc"
+            bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
         selector:
           matchLabels:
             app: smart-gateway
-  when: servicemonitor_manifest is not defined
+            smart-gateway: "{{ this_smartgateway }}"
 
 - name: Create ServiceMonitor to scrape Smart Gateway
+  k8s:
+    state: '{{ "present" if servicetelemetry_vars.backends.metrics.prometheus.enabled else "absent" }}'
+    definition:
+      '{{ sg_specific_servicemonitor_manifest }}'
+
+- name: Create additional serviceMonitor if provided
   k8s:
     state: '{{ "present" if servicetelemetry_vars.backends.metrics.prometheus.enabled else "absent" }}'
     definition:
       '{{ servicemonitor_manifest }}'
+  when: servicemonitor_manifest is defined
```

Comment:
Based on the comment above, if we moved to adding the `when: this_collector.servicemonitor_manifest not defined` in the above section, then the contents of the `servicemonitorManifest` would have already been dealt with by overriding the default `sg_specific_servicemonitor_manifest`, and this would simply be removed.